Adding thresholdType parameter to Calendars

Lokesh-Bharath opened the pull request:
# Description

Previously, we only allowed threshold's of type 'greaterThan' to maintain streaks in a calendar by default. This change adds a  new Parameter 'thresholdType' to let users choose between 'greaterThan' and 'lessThan' thresholds. The default value if the thresholdType parameter is not used still remains 'greaterThan'

Usecase Example - Now Users can set thresholds to break habits. Streak stays only if they played video games 'lessThan' 2 hours per day

Fixes # (issue)
#195, #246

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

- [x] If this is a new feature, did you add files to the examples directory to demonstrate the feature?
- [x] If this is a new feature, did you add documentation to the docs directory for the feature?

Lokesh-Bharath commented:
@lazyguru I've made the changes as we've discussed earlier. Please review them at your convinience! Also I had a couple of questions

1. I see testcases just to display the functionality and none to cover the success and failure cases. So added the testcase accordingly, let me know if I can add more cases anywhere.
2. Documentation was also just in one file Input Parameters, so I added the new parameter there, let me know if I'm missing something here!

lazyguru commented on examples/TestCalendar.md:
```diff
@@ -70,6 +70,26 @@ month:
+month:
+    startWeekOn:
+    threshold: 40
+    thresholdType: lessthan
```
In the docs and comments you use "LessThan", but in code it is expecting "lessthan". Can you please adjust one way or the other?

Lokesh-Bharath replied:
The code for handling this cases is **not case-sensitive**. It can handle the letters in any case as long as the spelling is the same. But to make it uniform and easier for users, I'll change it to "LessThan" in testcase as well and commit it in few minutes

Lokesh-Bharath commented:
@lazyguru Can we merge this if you feel the comments have been resolved? I'm ready to raise an other PR, It'll be easy to pull these changes (If this is good for merge) in and raise it to avoid conflicts

Lokesh-Bharath commented:
@lazyguru Let me know if you need any clarifications on this one

lazyguru commented:
Sorry. I know I owe you some reviews. I think I'm ok with the code here (I believe I reviewed it once before and you've addressed those comments?). I just have to find time to pull the branch down and manually verify things before I approve+merge.

Lokesh-Bharath commented:
Hey sorry I missed this message. Yes, I've addressed your comments. I did not raise the other PR yet (Although the code is commited in my fork) as it builds upon this changes (and it gets confusing). Once you find time to merge this, I can raise the other PR as well

lazyguru commented:
Sorry @Lokesh-Bharath, I'm going to get things caught up this week and will make time at least once a week to review+merge&deploy any new PRs. I checked this one and think there might be a problem on the "Multiple targets" test. I see results for PushUps, but if you click the title to toggle to "Meditation" there are no results. Maybe because they don't have values?

lazyguru commented:
Never mind, I double-checked and that is how it works on master also. So maybe still a bug, but not caused by these changes

Lokesh-Bharath commented:
Thank you, Joe!
It would be great if your schedule allows some time every week or two to review new changes. I completely understand how challenging it can be to maintain a project like this alongside personal and professional commitments. Really appreciate all the effort you're putting in to keep it going!
